pkg/handlers/helper.go

```diff
@@ -139,48 +137,46 @@ func alertName(a template.Alert) (*string, error) {
 	return nil, fmt.Errorf("no alertname defined in alert")
 }
 
-// SendServiceLog sends a servicelog notification for the given alert
-func (o *ocmsdkclient) SendServiceLog(summary, firingDesc, resolveDesc, clusterID string, severity v1alpha1.NotificationSeverity, logType string, references []v1alpha1.NotificationReferenceType, firing bool) error {
-	req := o.ocm.Post()
-	err := arguments.ApplyPathArg(req, "/api/service_logs/v1/cluster_logs")
+func (o *ocmsdkclient) SendServiceLog(summary, firingDesc, resolveDesc, clusterUUID string, severity v1alpha1.NotificationSeverity, logType string, references []v1alpha1.NotificationReferenceType, firing bool) error {
```

Comment:
What do you think of outsourcing the aggregation of those parameters to a `slv1.LogEntry` in a separate method and have the `SendServiceLog` method directly consume a `slv1.LogEntry`?

This is what I am doing in [my PR](https://github.com/openshift/ocm-agent/pull/68/files#diff-d8a640025cf97351b951865cfa2c487e92267cac2b507b583d6f35c056c1525fR28), except that I was still using type `ocm.ServiceLog` in place of `slv1.LogEntry`.

Indeed, with mocking in mind, `SendServiceLog` should do as less work as possible in order to get the unit tests cover as much code as possible.
Indeed in [my PR](https://github.com/openshift/ocm-agent/pull/68/files#diff-d8a640025cf97351b951865cfa2c487e92267cac2b507b583d6f35c056c1525fR55), I am adding the possibility to replace place holders with alert labels & annotations in the service log summary & description; this must be done **before** calling `SendServiceLog` as this method is designed to be mocked/replaced in unit tests.

Now of course, if `SendServiceLog` directly consumes a `slv1.LogEntry`, then it is probably just an empty wrapper around `serviceLogsHandler.PostServiceLog`. This may still have some value if we want to mock `SendServiceLog` but not mock `serviceLogsHandler.PostServiceLog`... however why not change the paradigm to say that it is `serviceLogsHandler.PostServiceLog` which should be mocked.

Reply:
Note that I can take care of changing the method signature while dealing with my change.

Reply:
Thanks you @Nikokolas3270 - To have the construction of `slv1.LogEntry` separated from the SendServiceLog function would indeed simplify the `SendServiceLog` function, making it easier to mock and test. 

However, considering your PR is in progress that may handle these changes, may be better I keep the current implementation for now to avoids conflicting changes?

Reply:
Hello, sure lets keep it that way, we can always come back on that later :)